File: leetcode/java/0232_implement_queue_using_stacks.java
class MyQueue {
    Stack<Integer> stk = new Stack<Integer>();

    public MyQueue() {
        
    }
    
    public void push(int x) {
        if (stk.size() == 0) {
            stk.push(x);
            return;
        }

        Stack<Integer> stk1 = new Stack<Integer>();
        while (!stk.isEmpty()) {
            stk1.push(stk.pop());
        }
        stk.push(x);
        while (!stk1.isEmpty()) {
            stk.push(stk1.pop());
        }
    }
    
    public int pop() {
        return stk.pop();
    }
    
    public int peek() {
        return stk.peek();
    }
    
    public boolean empty() {
        return stk.isEmpty();
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */

File: leetcode/cpp/0232_implement_queue_using_stacks.cpp
class MyQueue {
public:
    MyQueue() {

    }

    void push(int x) {
        if (stk1.empty()) {
            stk1.push(x);
            return;
        }

        while (!stk1.empty()) {
            stk2.push(stk1.top());
            stk1.pop();
        }

        stk1.push(x);

        while (!stk2.empty()) {
            stk1.push(stk2.top());
            stk2.pop();
        }
    }

    int pop() {
        int top = 0;

        if (!stk1.empty()) {
            top = stk1.top();
            stk1.pop();
        }
        
        return top;
    }

    int peek() {
        return stk1.top();
    }

    bool empty() {
        return stk1.empty();
    }

    stack<int> stk1, stk2;
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */